zcash_client_sqlite/src/wallet.rs

```diff
@@ -340,21 +366,91 @@ pub(crate) fn pool_code(pool_type: PoolType) -> i64 {
     }
 }
 
-pub(crate) fn scope_code(scope: Scope) -> i64 {
-    match scope {
-        Scope::External => 0i64,
-        Scope::Internal => 1i64,
+/// An enumeration of the scopes of keys that are generated by the `zcash_client_sqlite`
+/// implementation of the `WalletWrite` trait.
+///
+/// This extends the [`zip32::Scope`] type to include the custom scope used to generate keys for
+/// ephemeral transparent addresses.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum KeyScope {
+    /// A key scope corresponding to a [`zip32::Scope`].
+    Zip32(zip32::Scope),
+    /// An ephemeral transparent address, which is derived from an account's transparent
+    /// [`AccountPubKey`] with the BIP 44 path `change` level index set to the value `2`.
+    ///
+    /// [`AccountPubKey`]: zcash_primitives::legacy::keys::AccountPubKey
+    Ephemeral,
+}
+
+impl KeyScope {
```

Comment:
Why do we need another type rather than using `zcash_primitives::legacy::keys::TransparentKeyScope`?

Reply:
This scope type applies to both shielded and transparent addresses. While the encoding is shared, I do see the zip32 shielded scope type as logically distinct from the transparent key scope type.